PowerShellGet/PSModule.psm1

```diff
@@ -4906,6 +4906,16 @@ function Test-ScriptFileInfo
             return
         }
 
+        if ($PSScriptInfo.Version -match '-')
+        {
+            # Version contains PreRelease string.  Break apart and validate separately.
+            $indexOfFirstHyphen = $PSScriptInfo.Version.IndexOf('-')
+            $versionSubstring = $PSScriptInfo.Version.Substring(0, $indexOfFirstHyphen)
+            $preReleaseSubstring = $PSScriptInfo.Version.Substring($indexOfFirstHyphen+1)

```

Comment:
This logic works, there is no issue. I would like suggest you to try achieve this logic in PS scripting language elements like -split? If it is not possible to achieve in PS scripting, it is fine to use .Net API. The reason for this suggestion is to showcase PowerShellGet as the module developed in PS scripting. That was the one of the implementation goals of this module. There are some cases where we can't implement the required logic in PS, in those cases we are relying on .Net API :)